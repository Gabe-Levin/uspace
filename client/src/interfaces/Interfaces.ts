export interface UserSpaceRoleType {
  user: {
    username: string;
    email: string;
    picture_url?: string;
  };
}

export default interface SpaceWithCreatorType {
  id: number;
  name: string;
  created_at: Date;
  User_Space_Role: UserSpaceRoleType[];
}

export interface CommentType {
  id: number;
  content: string;
  created_at: string;
  post_id: number;
  user_id: number;
}

export interface PostType {
  id: number;
  content: string;
  space_id: 2;
  tags: string;
  title: string;
  user_id: string;
  created_at: Date;
  Comment: CommentType[];
}

export interface SpaceDataType {
  Post: PostType[];
  created_at: Date;
  description: string;
  id: number;
  name: string;
  User_Space_Role: UserSpaceRoleType[];
}

export interface UserType {
  id: number;
  email: string;
  email_verified: boolean;
  username: string;
  picture_url: string;
  sub: string;
  created_at: Date;
}
