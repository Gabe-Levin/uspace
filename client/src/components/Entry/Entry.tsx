import React, { useState } from 'react';
import './Entry.scss';
import { PostType } from '../../interfaces/Interfaces';

interface Incoming {
  post: PostType;
  setClickedPost: Function;
  index: number;
}

function Entry(props: Incoming) {
  const { post } = props;
  const [clicked, setClicked] = useState<boolean>(false);

  const changeDetails = () => {
    props.setClickedPost(props.index);
    setClicked((prev) => !prev); // TODO: only show one as active not two
  };

  // format date to show MONTH DAY
  const date = new Date(post.created_at).toLocaleDateString('en-EN', {
    month: 'short',
    day: 'numeric',
  });

  // remove html tags and only allow certain character amount
  const content =
    post.content.replace(/(<([^>]+)>)/gi, '').slice(0, 90) + '...';

  return (
    <div
      className={clicked ? 'entry active' : 'entry'}
      onClick={changeDetails}
    >
      <div className="entry-left">
        <div className="entry-left-date">{date}</div>
      </div>
      <div className="entry-right">
        <div className="entry-right-title">
          {post.title}
        </div>
        <div className="entry-right-text">{content}</div>
        {(post.Comment && post.Comment.length > 0) && (
          <div className="entry-right-tags">
            <div className="comment-tag">{post.Comment.length} comments</div>
            {/* <div className="attachment-tag">1 attachment</div> */}
          </div>
        )}
      </div>
    </div>
  );
}

export default Entry;
