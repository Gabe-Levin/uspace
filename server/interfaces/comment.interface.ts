export interface IComment {
  id: number,
  content: string,
  createdAt: Date,
  userId: number,
  postId: number,
}
