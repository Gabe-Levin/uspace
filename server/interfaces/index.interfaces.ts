type xy = {
  a: string
}

interface yz {
  a: string
}