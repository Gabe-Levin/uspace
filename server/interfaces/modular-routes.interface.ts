import { IComment } from '../interfaces/comment.interface';
import { IPost } from '../interfaces/post.interface';
import { IRole } from '../interfaces/role.interface';
import { ISpace } from '../interfaces/space.interface';
import { IUserSpaceRole } from '../interfaces/user-space-role.interface';
import { IUser } from '../interfaces/user.interface';
export type IAllTables = IComment | IPost | IRole | ISpace | IUserSpaceRole | IUser;
