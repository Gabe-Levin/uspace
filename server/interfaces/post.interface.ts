export interface IPost {
  id: number,
  title: string,
  tags: string[],
  createdAt: Date,
  content: string,
  userId: number,
  spaceId: number,
}
