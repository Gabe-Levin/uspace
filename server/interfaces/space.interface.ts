import { Space } from '@prisma/client';

export interface ISpace{
  id: number,
  name: string,
  description: string,
  createdAt: Date,
}

export interface IIncomingSpace extends Omit<Space, 'id'|'created_at'>{}
