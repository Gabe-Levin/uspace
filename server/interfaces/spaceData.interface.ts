import { IComment } from './comment.interface';
import { IPost } from './post.interface';
import { ISpace } from './space.interface';

export interface IStrippedUser {
  username: string, email:string, pictureUrl: string, id: number,
}

export interface IPostWithComments extends IPost{
  comments: IComment[],
}
export interface ISpaceData extends ISpace {
    posts: IPostWithComments[],
    userSpaceRoles: {user: IStrippedUser}[],
}
