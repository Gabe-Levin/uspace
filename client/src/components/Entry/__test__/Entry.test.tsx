import { cleanup } from '@testing-library/react';
import renderer from 'react-test-renderer';
import { IncomingEntry, PostType } from '../../../interfaces/Interfaces';
import Entry from '../Entry';

describe('Testing Entry component', () => {

  const MOCK_POST: PostType={
    id: 1,
    content: "testContent",
    tags: "testTags",
    title: "testTitle",
    userId: 1,
    spaceId: 1,
  } as PostType

  const MOCK_PROPS: IncomingEntry={
    post: MOCK_POST,
    setClickedPost: jest.fn(),
    index: 1,
    activePostId: 1,
    setActivePostId: jest.fn()
  }

  afterEach(cleanup);
  it('Matches snapshot', () => {
    const tree = renderer.create(<Entry 
      index={MOCK_PROPS.index}
      post={MOCK_PROPS.post}
      setClickedPost={jest.fn()}
      activePostId={MOCK_PROPS.activePostId}
      setActivePostId={MOCK_PROPS.setActivePostId} />).toJSON();
    expect(tree).toMatchSnapshot();
  })
})