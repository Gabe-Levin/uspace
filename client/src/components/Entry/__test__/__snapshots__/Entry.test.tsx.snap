// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing Entry component Matches snapshot 1`] = `
<div
  className="entry active"
  onClick={[Function]}
>
  <div
    className="entry-left"
  >
    <div
      className="entry-left-date"
    >
      Invalid Date
    </div>
  </div>
  <div
    className="entry-right"
  >
    <div
      className="entry-right-title"
      dangerouslySetInnerHTML={
        Object {
          "__html": "testTitle",
        }
      }
    />
    <div
      className="entry-right-text"
    >
      testContent...
    </div>
  </div>
</div>
`;
